docs/tutorial/composition.md

```diff
@@ -0,0 +1,219 @@
+|     7     || 8         | 8         |
+|     8     || 9         | 9         |
+|     9     || 10        | 10        |
+```
+
+## Multi inputs, multi outputs
+
+Composition is not limited to 1-to-1 circuits, you can also use it with circuits with multiple inputs and multiple outputs. Here is an example that computes the 10 first elements of the Fibonacci sequence in fhe:
+
+```python
+from concrete import fhe
+
+def identity(x):
+   return fhe.univariate(lambda x: x)(x)
+
+@fhe.compiler({"n1th": "encrypted", "nth": "encrypted"})
+def fib(n1th, nth):
+   return identity(nth), identity(n1th + nth)
+
+print("Compiling `fib` function ...")
+inputset = list(zip(range(0, 100), range(0, 100)))
+fib_fhe = fib.compile(inputset, composable=True)
+
+print("Generating keyset ...")
+fib_fhe.keygen()
+
+print("Encrypting initial values")
+n1th = 1
+nth = 2
+(n1th_enc, nth_enc) = fib_fhe.encrypt(n1th, nth)
+
+print(f"|           ||        (n-1)-th       |         n-th          |")
+print(f"| iteration || decrypted | cleartext | decrypted | cleartext |")
+for i in range(10):
+   (n1th_enc, nth_enc) = fib_fhe.run(n1th_enc, nth_enc)
+   (n1th, nth) = fib(n1th, nth)
+   
+    # For demo purpose; no decryption is needed.
+   (n1th_dec, nth_dec) = fib_fhe.decrypt(n1th_enc, nth_enc)
+   print(f"|     {i}     || {n1th_dec:<9} | {n1th:<9} | {nth_dec:<9} | {nth:<9} |")
+```
+
+Executing this script should provide the following output:
+
+```shell
+Compiling `fib` function ...
+Generating keyset ...
+Encrypting initial values
+|           ||        (n-1)-th       |         n-th          |
+| iteration || decrypted | cleartext | decrypted | cleartext |
+|     0     || 2         | 2         | 3         | 3         |
+|     1     || 3         | 3         | 5         | 5         |
+|     2     || 5         | 5         | 8         | 8         |
+|     3     || 8         | 8         | 13        | 13        |
+|     4     || 13        | 13        | 21        | 21        |
+|     5     || 21        | 21        | 34        | 34        |
+|     6     || 34        | 34        | 55        | 55        |
+|     7     || 55        | 55        | 89        | 89        |
+|     8     || 89        | 89        | 144       | 144       |
+|     9     || 144       | 144       | 233       | 233       |
+```
+
+Though it is not visible in this example, you don't need to have a matching number of inputs and outputs. Nor do you need to have a specific logic in mind for how you forward values from outputs to inputs; those can be switched for instance. 
+
+{% hint style="info" %}
+If you wonder why we wrapped the outputs into `identity`, see below in the [Gotcha section](#gotcha).
+{% endhint %}
+
+## Iteration support
+
+In the previous example you can see that to some extent, composition allows to support iteartion with cleartext iterands. That is loops with shape :
+
+```python
+for i in some_cleartext_range:
+    # Do something in FHE in the loop body, implement as an fhe circuit.
```

Comment:
I would also write "FHE" in the comments

Reply:
I don't understand what you mean here.

Reply:
I mean: capitalize FHE please. There is an 'fhe' in '# Do something in FHE in the loop body, implement as an fhe circuit.'

Reply:
I missed the second one. Got it.